test/e8.test.js

```diff
@@ -1,32 +1,49 @@
-import { vitest, it, describe, expect, } from "vitest";
-import { promiseResolved, promiseRejected } from "../exercises/e8.js";
+import { it, describe, expect, beforeEach, afterEach } from "vitest";
+import { returnPromise, returnRejectingPromise } from "../exercises/e8.js";
 
-describe("promiseResolved", () => {
-  it("promiseResolved variable should exist", () => {
-    expect(promiseResolved).toBeInstanceOf(Object);
+describe("returnPromise", () => {
+  it("returnPromise variable should exist", () => {
+    expect(returnPromise).toBeInstanceOf(Function);
   });
 
-  it("promiseResolved constructor should have a name of Promise", () => {
-    expect(promiseResolved.constructor.name).toEqual("Promise");
+  it("returnPromise should return a Promise", () => {
+    expect(returnPromise().constructor.name).toEqual("Promise");
   });
 
-  it("Promise should log a resolved message in the console", async () => {
-    const logSpy = vitest.spyOn(console, "log");
-    await promiseResolved.then((d) => console.log(d));
-    expect(logSpy).toHaveBeenLastCalledWith("The PROMISE was RESOLVED");
+  it("returnPromise should resolve to 'The PROMISE was RESOLVED'", async () => {
+    expect(await returnPromise()).toBe("The PROMISE was RESOLVED");
+  });
+
+  it("returnPromise should have less than `` characters", () => {
```

Comment:
The value of the characters is missing in the test description.

With the "reject" argument should be less than 82,
Without the "reject"  (since resolves only) argument should be less than 72 characters.

I believe the less than 60 characters should be a fine value since the final result of const returnPromise = () => Promise.resolve('The PROMISE was RESOLVED'); is 49 characters in length.

Reply:
This might be a personal preference but to me the following test description would make a bit more sense:

"The string representing the source code of the `returnPromise` function should have less than `60` characters"

Googling for "string representing the source code of a function"  provides the first link in search results to Function.prototype.toString()

Reply:
Resolved.